File: Projekt_Python/rosliny/Guarana.py
from Roslina import Roslina

class Guarana (Roslina): 
    def __init__(self, x, y,dane):
        if dane==None:
            self._x = x
            self._y = y
            self._symbol = 'g'
            self._sila = 0
            self._wykonal_ruch = False
            self._inicjatywa = 0
            self._zycie = 0
            self._numer=2
        else:
            self._x = dane[0]
            self._y = dane[1]
            self._inicjatywa = dane[2]
            self._symbol = 'g'
            self._sila = dane[3]
            self._wykonal_ruch = False
            self._zycie = dane[4]
            self._numer = dane[5]
        self._kolor= (255,0,0)

    def kolizja(self, organizm, organizmy, j, swiat):
        organizm.setSila(organizm.getSila()+3)
        swiat.setRaport(self._symbol+" ginie od "+ organizm.getSymbol())
        organizmy.remove(self)
        swiat.setPlanszaKolizja(self._x,self._y,organizm)

File: projekt_java/src/po/java/rosliny/Guarana.java
package po.java.rosliny;

import po.java.Organizm;
import po.java.Roslina;
import po.java.Swiat;

import java.awt.*;
import java.util.ArrayList;

public class Guarana extends Roslina {
    public Guarana(int x, int y) {
        this.x = x;
        this.y = y;
        this.symbol = 'g';
        this.sila = 0;
        this.wykonal_ruch = false;
        this.zycie = 0;
        this.numer=2;
        this.kolor= Color.RED;
    }
    public Guarana(int []dane)
    {
        this.x = dane[0];
        this.y = dane[1];
        this.inicjatywa=dane[2];
        this.symbol = 'g';
        this.sila = dane[3];
        this.wykonal_ruch = false;
        this.zycie = dane[4];
        this.numer=dane[5];
        this.kolor= Color.RED;
    }
    @Override
    public void kolizja(Organizm organizm, ArrayList organizmy, int j, Swiat swiat)
    {
        organizm.setSila(organizm.getSila()+3);
        swiat.setRaport(symbol+" ginie od "+ organizm.getSymbol());
        organizmy.remove(this);
        swiat.setPlanszaKolizja(x,y,organizm);
    }
}
